### PR TITLE
Add no lm support for CTC decoder

### PR DESCRIPTION
**Pull request by carolineechen**
Add support for CTC lexicon decoder without LM support by adding a non language model `ZeroLM` that returns score 0 for everything. Generalize the decoder class/API a bit to support this, adding it as an option for the kenlm decoder at the moment (will likely be separated out from kenlm when adding support for other kinds of LMs in the future)

### CODE REVIEW

File: torchaudio/prototype/ctc_decoder/ctc_decoder.py [hunk 1]
```diff
@@ -30,8 +32,8 @@ class Hypothesis(NamedTuple):
     score: float
 
 
-class KenLMLexiconDecoder:
-    """torchaudio.prototype.ctc_decoder.KenLMLexiconDecoder()
+class LexiconDecoder:
+    """torchaudio.prototype.ctc_decoder.LMLexiconDecoder()
```
Comment (hwangjeff) [inline torchaudio/prototype/ctc_decoder/ctc_decoder.py:36]:
```suggestion
    """torchaudio.prototype.ctc_decoder.LexiconDecoder()
```

and similar for the docstring for `Hypothesis`

File: torchaudio/prototype/ctc_decoder/ctc_decoder.py [hunk 2]
```diff
@@ -185,7 +187,7 @@ def kenlm_lexicon_decoder(
             Each line consists of a word and its space separated spelling
         tokens (str or List[str]): file or list containing valid tokens. If using a file, the expected
             format is for tokens mapping to the same index to be on the same line
-        kenlm (str): file containing languge model
+        kenlm (str): file containing languge model, or empty string if not using a language model
```
Comment (hwangjeff) [inline torchaudio/prototype/ctc_decoder/ctc_decoder.py:190]:
```suggestion
        kenlm (str): file containing language model, or empty string if not using a language model
```

File: torchaudio/csrc/decoder/src/decoder/lm/ZeroLM.h [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#pragma once
+
+#include "torchaudio/csrc/decoder/src/decoder/lm/LM.h"
+
+namespace torchaudio {
+namespace lib {
+namespace text {
+
+/**
+ * ZeroLM is a dummy language model class, which mimics the behavious of a
```
Comment (hwangjeff) [inline torchaudio/csrc/decoder/src/decoder/lm/ZeroLM.h:17]:
```suggestion
 * ZeroLM is a dummy language model class, which mimics the behavior of a
```

**Review by hwangjeff — APPROVED** [review]

File: torchaudio/prototype/ctc_decoder/ctc_decoder.py [hunk 4]
```diff
@@ -12,14 +13,15 @@
...
 )
 
 
-__all__ = ["Hypothesis", "KenLMLexiconDecoder", "kenlm_lexicon_decoder"]
+__all__ = ["Hypothesis", "LexiconDecoder", "kenlm_lexicon_decoder"]
 
 
 class Hypothesis(NamedTuple):
-    r"""Represents hypothesis generated by CTC beam search decoder :py:func`KenLMLexiconDecoder`.
+    r"""Represents hypothesis generated by CTC beam search decoder :py:func`LMLexiconDecoder`.
```
Comment (carolineechen) [inline torchaudio/prototype/ctc_decoder/ctc_decoder.py:24]:
```suggestion
    r"""Represents hypothesis generated by CTC beam search decoder :py:func`LexiconDecoder`.
```

File: torchaudio/prototype/ctc_decoder/ctc_decoder.py [hunk 5]
```diff
@@ -163,7 +165,7 @@ def idxs_to_tokens(self, idxs: torch.LongTensor) -> List:
 def kenlm_lexicon_decoder(
```
Comment (nateanl) [inline torchaudio/prototype/ctc_decoder/ctc_decoder.py:165]: Maybe rename the `kenlm_lexicon_decoder` to `lexicon_decoder` to cover the empty lm case, what do you think?

Comment (mthrok) [reply torchaudio/prototype/ctc_decoder/ctc_decoder.py:165]: I think that makes sense.

Also I think that the namespace can be `torchaudio.prototype.decoder`. (not the scope of this PR though)

**Review by mthrok — APPROVED** [review]

File: examples/tutorials/asr_inference_with_ctc_decoder_tutorial.py [hunk 6]
```diff
@@ -171,6 +171,9 @@
 # In addition to the previously mentioned components, it also takes in various beam
 # search decoding parameters and token/word parameters.
 #
+# This decoder can also be run without a language model by passing in the empty string `""`
```
Comment (mthrok) [inline examples/tutorials/asr_inference_with_ctc_decoder_tutorial.py:174]: This line also needs to be updated.

File: torchaudio/prototype/ctc_decoder/ctc_decoder.py [hunk 7]
```diff
@@ -185,7 +187,7 @@ def kenlm_lexicon_decoder(
             Each line consists of a word and its space separated spelling
         tokens (str or List[str]): file or list containing valid tokens. If using a file, the expected
             format is for tokens mapping to the same index to be on the same line
-        kenlm (str): file containing languge model
+        lm (Optional[str], optional): file containing language model, or `None` if not using a language model
```
Comment (nateanl) [inline torchaudio/prototype/ctc_decoder/ctc_decoder.py:190]:
```suggestion
        lm (str or None, optional): file containing language model, or `None` if not using a language model
```

**Review by nateanl — APPROVED** [review]
LGTM. Just a nit of doc string.

**Pull request closed without merging**